pkg/secureenclave/secureenclave.go

```diff
@@ -0,0 +1,141 @@
+//go:build darwin
+// +build darwin
+
+package secureenclave
+
+// this was heavily inspiried by https://github.com/facebookincubator/sks
+// thank you!
+
+/*
+#cgo darwin LDFLAGS: -framework Foundation -framework Security -framework CoreFoundation
+#include <stdlib.h>
+#include <secureenclave.h>
+*/
+import "C"
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha1"
+	"errors"
+	"fmt"
+	"io"
+	"unsafe"
+)
+
+type SecureEnclaveSigner struct {
+	publicKey *ecdsa.PublicKey
+}
+
+// New verifies that the provided public key already exists in the secure enclave.
+// Then returns a new Secure Enclave Keyer using the provided public key.
+func New(publicKey ecdsa.PublicKey) (*SecureEnclaveSigner, error) {
```

Comment:
I suspect that this should take the raw `[]byte`, and that `CreateKey` should return that. Same as the tpm.

Reply:
done